enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,359 @@
+
+    // networkLoadBalancerParameters holds configuration parameters for an AWS
+    // network load balancer. Present only if type is NLB.
+    //
+    // +optional
+    NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
+}
+
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network load balancer.
+type AWSNetworkLoadBalancerParameters struct {
+    EIPAllocations EIPAllocations `json:"eip-allocations,omitempty"`
+}
+
+type EIPAllocations string
+
+```
+
+- The API field for `eip-allocations` field in the Ingress Controller CR needs to be set as follows
+  in the file `operator/v1/types_ingress.go`:
+```go
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network load balancer.
+type AWSNetworkLoadBalancerParameters struct {
+    EIPAllocations EIPAllocations `json:"eip-allocations,omitempty"`        
+}
+
+type EIPAllocations string
+
+```
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Currently, there is no unique consideration for making this change work with Hypershift.
+
+#### Standalone Clusters
+
+Is the change relevant for standalone clusters?
+
+#### Single-node Deployments or MicroShift
+
+This proposal does not affect the resource consumption of a
+single-node OpenShift deployment (SNO), CPU and memory.
+
+This proposal does not affect MicroShift. The proposal
+adds configuration options through API resources and the installer however, it is not
+required for any of those behaviors to be exposed to MicroShift admins through the
+configuration file for MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+#### 1. Set EIP through installer for the default ingress controller:
+   
+   a. The admin sets the `eip-allocations` in the installer using the install-config.yaml.
+   
+   b. The installer then sets the ingress cluster object with the `eip-allocations`.
+   
+   c. The cluster ingress operator then creates a default ingress controller by:
+
+     i.   creating an default ingress controller CR
+     
+     ii.  then scaling a deployment for the Ingress Controller CR which has the value of the `eip-allocations`.
+     
+     iii. then creating a service with service type load balancer with annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+          which uses the value from the field `eip-allocations` of ingress controller CR.
+
+#### 2. Set EIP through the Ingress Controller CRD:
+   
+   The admin sets the `eip-allocations` in the Ingress Controller CR. Then same process
+   as per point 1.c.ii and 1.c.iii is followed by the ingress operator. 
+
+#### 3. Update the existing EIP for the IngressController:
+   
+   When the admin changes the eip-allocations field in the Ingress Controller CR, the watch on the
+   `eip-allocations` field by the ingress operator scales a new deployment for the Ingress Controller CR.
```

Comment:
Oh, so if you change the EIP allocations then you get a "brand new" ingresscontroller? That might be surprising as it will drop all traffic. Seems subtle.

Reply:
Yeah, after testing I saw we can't get the annotations changed on the fly. If someone wants to update the eip-allocations  then service will have to deleted and then recreated with the new value for `eip-allocations` field.

Reply:
@frobware I need to test again. It looks like if the service is updated with the annotation the kcm should update the eips. https://github.com/brooksgarrett/kubernetes/blob/72dd54ec4856ccb7bd2b08258d95f3d0b932fb11/pkg/cloudprovider/providers/aws/aws.go#L4157

Reply:
It looks like https://github.com/openshift/enhancements/compare/45ad01df72b72174f9347fcb54179af4661cbd19..1137a8acba5a4c72ae496c34fa93b97dd18d3990 resolved this question by removing "scales a new deployment" and instead describing the workflow for recreating the service.